crates/arrow_tools/src/format.rs

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use arrow::array::{Array, ArrayRef, FixedSizeListArray, ListArray, StructArray};
+use arrow::buffer::OffsetBuffer;
+use arrow::compute::concat;
+use arrow_schema::{ArrowError, DataType, Field};
+use std::sync::Arc;
+
+/// Operations to apply to [`ArrayRef`] or [`RecordBatch`] data so as to prepare it for display.
+///
+/// Note: Operations do not preserve all original data, and as such, should be used for human display purposes only.
+pub enum FormatOperation {
+    /// Truncate strings to be no larger than a given length. If `bool=true`, then Utf8 elements within lists and structs will be truncated.
+    TruncateUtf8Length(usize, bool),
```

Comment:
Why not just always truncate Utf8 within lists and structs?

Reply:
And I would make this a separate variant instead of a bool in `TruncateUtf8Length` if you wanted a separate behavior for truncating strings within lists/structs